certs-manager/src/main/java/io/strimzi/certs/OpenSslCertManager.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017-2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.certs;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * An OpenSSL based certificates manager
+ */
+public class OpenSslCertManager implements CertManager {
+
+    private static final Logger log = LogManager.getLogger(OpenSslCertManager.class);
+
+    @Override
+    public void generateSelfSignedCert(File keyFile, File certFile, int days) throws IOException {
+        generateSelfSignedCert(keyFile, certFile, null, days);
+    }
+
+    @Override
+    public void generateSelfSignedCert(File keyFile, File certFile, Subject sbj, int days) throws IOException {
```

Comment:
I think we might need to support setting the alternative subject names as mentioned by Rob in Norway. Not sure completely sure whether it should be standalone param or part of the Subject in this impl.

Reply:
Yeah right. I would have it in the `Subject` entity. Interesting that the openssl tool doesn't provide a way to pass them on the command line but writing a sort of configuration file with `subjectAltName` is needed.

Reply:
After "fighting" with openssl I developed that but it seems to work pretty fine with self signed certificate. Instead, when a csr is created and it contains subject alt names (in the x509 extensions section), these extensions are not copied by default from the csr to the final certificate during the signing process. It's related to the `copy_extensions` option which is false by default and not available on the `x509` command option (https://github.com/openssl/openssl/issues/3638)

Reply:
Instead of the `x509` command for signing a certificate from a CR it seems the `ca` command is available but I should understand the differences between them first.